lib/api/src/modules/refs.ts

```diff
@@ -135,28 +135,27 @@ export const init: ModuleFn = ({ store, provider, fullAPI }, { runCheck = true }
       });
     },
     checkRef: async (ref) => {
-      const { id, url, version } = ref;
+      const { id, url, version, type } = ref;
+      const isPublic = type === 'server-checked';
 
       const loadedData: { error?: Error; stories?: StoriesRaw; loginUrl?: string } = {};
       const query = version ? `?version=${version}` : '';
 
-      const [included, omitted, iframe] = await allSettled([
-        fetch(`${url}/stories.json${query}`, {
-          headers: {
-            Accept: 'application/json',
-          },
-          credentials: 'include',
-        }),
+      const [included, omitted] = await allSettled([
+        isPublic
+          ? Promise.resolve(false)
+          : fetch(`${url}/stories.json${query}`, {
+              headers: {
+                Accept: 'application/json',
+              },
+              credentials: 'include',
+            }),
         fetch(`${url}/stories.json${query}`, {
           headers: {
             Accept: 'application/json',
           },
           credentials: 'omit',
         }),
```

Comment:
If we know `iframe.html` fails without credentials (i.e. `isPublic == false`), the second request is definitely going to fail.

I recommend simplifying to:
```suggestion
      // In theory the `/iframe.html` could be private and the `stories.json` could still exist, but in practice
      // the only private servers we know about (Chromatic) always include `stories.json`. So we can tell
      // if the ref actually exists by simply checking `stories.json` w/ credentials.
      const storiesJsonAvailable = 
        await fetch(`${url}/stories.json${query}`, {
          headers: {
            Accept: 'application/json',
          },
          // If we include credentials the server needs proper CORS headers, so don't include them
          // unless we know we need them (i.e. initial fetch of `/iframe.html` failed).
          credentials: isPublic ? 'omit' : 'include',
        }));
```